crates/logic-test/src/db/gluesql.rs

```diff
@@ -19,163 +8,328 @@ use crate::{
 use super::{Execute, Output};
 
 pub struct GlueSQL {
-    storage: Glue<MemoryStorage>,
+    storage: Option<Glue<MemoryStorage>>,
 }
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    #[error(transparent)]
+    #[error("gluesql error: {0}")]
     GlueSQL(#[from] gluesql::prelude::Error),
 }
 
+impl sqllogictest::DB for GlueSQL {
```

Comment:
Is there any particular reason for not implementing `AsyncDB`? 
It seems possible if the `Glue` instance is created inside the `run` function and the `GlueSQL` struct holds the state using something like `Arc<Mutex<MemoryStorage>>`.